chart/templates/workers/worker-hpa.yaml

```diff
@@ -0,0 +1,49 @@
+{{/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+*/}}
+
+################################
+## Airflow Worker HPA
+#################################
+{{- if and (and (not .Values.workers.keda.enabled) .Values.workers.hpa.enabled) (has .Values.executor (list "CeleryExecutor" "CeleryKubernetesExecutor")) }}
+apiVersion: autoscaling/v2
```

Comment:
Per my knowledge, autoscaling/v2 is only available for v1.23 K8S and up. We need to add a conditional here.

Reply:
Thanks for reviewing @amoghrajesh. In a [previous PR](https://github.com/apache/airflow/pull/28461) the kubeVersion checker was removed since minimum supported version is 1.23. Should we still add this check to HPA?

Reply:
Agree we do not have to . We have now min k8s version even 1.25 - see https://github.com/apache/airflow?tab=readme-ov-file#requirements so no extra check is needed.